### PR TITLE
set client id in call options

### PR DESCRIPTION
**Pull request by cdcarter**
For all SimpleSalesforce instances we summon up, set the Sforce-Call-Options header to include a client id when registered or the CumulusCI clientName.

### CODE REVIEW

File: cumulusci/tasks/salesforce/BaseSalesforceApiTask.py [hunk 1]
```diff
@@ -28,6 +28,9 @@ def _init_api(self, base_url=None):
         )
         if base_url is not None:
             rv.base_url += base_url
+
+        rv.headers.setdefault(CALL_OPTS_HEADER_KEY, self._get_client_name())
```
Comment (davisagli) [inline cumulusci/tasks/salesforce/BaseSalesforceApiTask.py]: Should be in the form `client=CLIENT_NAME`, not simply the CLIENT_NAME

Comment (cdcarter) [reply cumulusci/tasks/salesforce/BaseSalesforceApiTask.py]: I woke up, hopped out of bed, and realized that this wasn't correct :)

Comment (cdcarter) [reply cumulusci/tasks/salesforce/BaseSalesforceApiTask.py]: I don't know what that says about me.

**Comment by cdcarter** [comment]
API Calls will be grouped properly when reporting.

**Comment by cdcarter** [comment]
And yea, it is a little weird that its a header and its mostly undocumented behavior...

**Pull request merged**